src/resources/filters/customnodes/floatreftarget.lua

```diff
@@ -787,9 +787,44 @@ end, function(float)
       float.identifier)
   end
 
-  float.caption_long.content:insert(1, pandoc.RawInline("asciidoc", ". "))
+  if float.type == "Table" and float.content.t == "Table" then
+    -- special-case the situation where the figure is Table and the content is Table
+    --
+    -- just return the table itself with the caption inside the table
+    float.content.caption.long = float.caption_long
+    float.content.attr = pandoc.Attr(float.identifier, float.classes or {}, float.attributes or {})
+    return pandoc.Blocks({
+      pandoc.RawBlock("asciidoc", "[[" .. float.identifier .. "]]\n"),
+      float.content
+    })
+  end
+
+  if float.type == "Figure" then
+    local count = 0
+    local img
+    _quarto.ast.walk(float.content, {
+      Image = function(node)
+        count = count + 1
+        img = node
+      end
+    })
+    if count == 1 then
```

Comment:
This is to let subfigures pass through right ?

Reply:
It's to make this fix as narrow as possible.

Reply:
ok I see. Thanks